rdd/utils.py

```
@@ -126,6 +143,255 @@ def _validate_groups(
+    ValueError
+        If both raw_gnps_network and external_sample_metadata are None.
+    """
+
+    if external_sample_metadata:
+
+        if not external_sample_metadata.lower().endswith(
+            (".csv", ".tsv", ".txt")
+        ):
+            raise ValueError(
+                "External metadata file must be a CSV, TSV, or TXT."
+            )
```

Ruff (0.14.1)

Avoid specifying long messages outside the exception class

(TRY003)